common/status-effects/weaknessRef.ts

```diff
@@ -0,0 +1,148 @@
+import StatusEffect from './status-effect'
+import {GameModel} from '../models/game-model'
+import {CARDS, HERMIT_CARDS} from '../cards'
+import {getCardPos, CardPosModel} from '../models/card-pos-model'
+import {getActiveRow, applyStatusEffect, removeStatusEffect} from '../utils/board'
+import {StatusEffectT, CardT} from '../types/game-state'
+import {STRENGTHS} from '../const/strengths'
+
+class WeaknessStatusEffect extends StatusEffect {
+	constructor() {
+		super({
+			id: 'weakness',
+			name: 'Weakness',
+			description: 'This will assign dummies for UI.',
+			duration: 3,
+			counter: false,
+			damageEffect: false,
+			visible: false,
+		})
+	}
+
+	override onApply(game: GameModel, statusEffectInfo: StatusEffectT, pos: CardPosModel) {
+		game.state.statusEffects.push(statusEffectInfo)
+		const {player, opponentPlayer} = pos
+		const strongType = HERMIT_CARDS[getActiveRow(opponentPlayer)!.hermitCard.cardId].hermitType
+		const weakType = HERMIT_CARDS[getActiveRow(player)!.hermitCard.cardId].hermitType
+
+		// Purely asthetic.
+		function applyDummies() {
+			for (let i = 0; i < player.board.rows.length; i++) {
+				const row = opponentPlayer.board.rows[i]
+
+				if (row.hermitCard) {
+					const card = row.hermitCard
+					if (HERMIT_CARDS[card.cardId].hermitType == weakType) {
+						applyStatusEffect(
+							game,
+							'weaknessdummy',
+							card.cardInstance,
+							Math.random().toString(),
+							statusEffectInfo.duration!
+						)
+					}
+				}
+			}
+			for (let i = 0; i < player.board.rows.length; i++) {
+				const row = player.board.rows[i]
+
+				if (row.hermitCard) {
+					const card = row.hermitCard
+					if (HERMIT_CARDS[card.cardId].hermitType == weakType) {
+						applyStatusEffect(
+							game,
+							'weaknessdummy',
+							card.cardInstance,
+							'receiverWeakness',
+							statusEffectInfo.duration!
+						)
+					}
+				}
+			}
+		}
+
+		if (!statusEffectInfo.duration) statusEffectInfo.duration = this.duration
+
+		applyDummies()
+
+		STRENGTHS[strongType].push(weakType)
```

Comment:
This doesn't work. This modifies the weak types for everyone changing the game.
This variable is supposed to be immutable.